tez-mapreduce/src/main/java/org/apache/tez/mapreduce/client/YARNRunner.java

```diff
@@ -623,6 +623,7 @@ public JobStatus submitJob(JobID jobId, String jobSubmitDir, Credentials ts)
     try {
       dagAMConf.set(TezConfiguration.TEZ_AM_STAGING_DIR,
           jobSubmitDir);
+      dagAMConf.setBoolean(TezConfiguration.TEZ_AM_STAGING_BASE_DIR_CLEANUP, true);
```

Comment:
YARNRunner is a glue code between a MapReduce Job and Tez, implementing `ClientProtocol`. So, the client code of YARNRunner is Apache Hadoop.
`ClientProtocol` doesn't have an API to declare that a specific job has been completed. If we resolve this issue on the client side, we have to add new APIs to Apache Hadoop. That's why I added a new param and handled the issue on Apache Tez side.
I'm not confident that this approach is the best. I'd appreciate it if someone could give me a better idea.